tests/OtpTest.php

```diff
@@ -110,14 +110,62 @@ public function testTotpRfc()
 		$this->assertEquals('47863826', $this->Otp->hotp($secret, floor(20000000000/30)), 'sha512 with time 20000000000');
 		*/
 	}
-	
+	public function invalidCounterValues()
+	{
+		return [['a'], [-1]];
+	}
+
+	/**
+	 * @dataProvider invalidCounterValues
+	 * @expectedException InvalidArgumentException
+	 * @expectedExceptionMessage Invalid counter supplied
+	 */
+	public function testHotpInvalidCounter($counter)
+	{
+		$this->Otp->hotp($this->secret, $counter);
+	}
+
+	/**
+	 * Tests Otp->checkHotpResync()
+	 */
+	public function testHotpResync()
+	{
+		$secret = $this->secret;
+
+		// test default counter window
+		$this->assertSame(0, $this->Otp->checkHotpResync($secret, 0, '755224'));
```

Comment:
I converted all the tests to use data sets. 
Yes this will definitely be helpful in the future. Now I know more about unit testing :)